sweyer_plugin/lib/sweyer_plugin.dart

```diff
@@ -0,0 +1,183 @@
+import 'dart:typed_data';
+import 'dart:ui';
+
+import 'package:equatable/equatable.dart';
+import 'package:sweyer_plugin/sweyer_plugin_method_channel.dart';
+import 'package:uuid/uuid.dart';
+
+import 'sweyer_plugin_platform_interface.dart';
+
+/// A song from the native media store.
+abstract class MediaStoreSong {
+  /// The id of this song in the native media store.
+  int get sourceId;
+
+  /// Absolute filesystem path to the media item on disk.
+  String? get filesystemPath;
+}
+
+/// An album from the native media store.
+abstract class MediaStoreAlbum {}
+
+/// An artist from the native media store.
+abstract class MediaStoreArtist {}
+
+/// A playlist from the native media store.
+abstract class MediaStorePlaylist {
+  /// The id of this playlist in the native media store.
+  int get id;
+
+  /// A list of ids of songs that this playlist contains.
+  List<int> get songIds;
+}
+
+/// A genre from the native media store.
+abstract class MediaStoreGenre {}
+
+abstract class SweyerPluginException with EquatableMixin {
```

Comment:
All the abstract interfaces should go into the sweyer_plugin_platform_interface.dart file, which represents a core platform independent abstraction.

To better understand why let me refer to the documentation https://docs.flutter.dev/packages-and-plugins/developing-packages#federated-plugins

In our case:
- `sweyer_plugin.dart` is like the **app-facing package**
- `android` folder is like the **platform package**
- `sweyer_plugin_method_channel.dart` and `sweyer_plugin_platform_interface.dart` is the **platform interface**

So, with the current location of this class, we mix these layers in between, because now the **platform interface** (SweyerMethodChannelException) extends from something that is located in the app facing package

Reply:
Hm, the problem is that these exceptions are app facing, the app wants to catch them and react to them.

So I guess the problem is that we are bringing method channel exceptions pretty much 1:1 into the app facing interface. Do you think defining separate exceptions in the `sweyer_plugin.dart` and then converting from method channel exceptions into those in the `sweyer_plugin_method_channel.dart` would be better?

Reply:
> So I guess the problem is that we are bringing method channel exceptions pretty much 1:1 into the app facing interface. Do you think defining separate exceptions in the sweyer_plugin.dart and then converting from method channel exceptions into those in the sweyer_plugin_method_channel.dart would be better?

No, that's not a problem, the composition is fine.

I was only suggesting that you move it into the `sweyer_plugin_platform_interface.dart`, still use in the `sweyer_plugin_method_channel.dart` (there is no problem with that), and then export from the app-facing file.

The problem is that down layer (method channel) currently uses something from the upper layer (app-facing package), which shouldn't happen, it should be vice-versa.

Reply:
You are right, it makes more sense that way. I still felt decoupling the `PlatformException` a bit more from the `SweyerExeption` would make the interface a bit nicer. Let me know if that is ok for you.